### PR TITLE
ZCS-10642: SearchGal | add support for query in request

### PR DESCRIPTION
**Pull request by itsmeprashant**
Added query param in searchGal request to support searching based on queries. This is needed for the organisation chart zimlet.

### CODE REVIEW

File: store/src/java/com/zimbra/cs/gal/GalSearchControl.java [hunk 1]
```diff
@@ -346,12 +346,15 @@ private void generateSearchQuery(Account galAcct) throws ServiceException {
     private String getSearchQuery(Account galAcct, boolean addInId) throws ServiceException {
         String query = mParams.getQuery();
         String searchByDn = mParams.getSearchEntryByDn();
+        String galSearchQuery = mParams.getGalSearchQuery();
```
Comment (dasiyogesh) [inline store/src/java/com/zimbra/cs/gal/GalSearchControl.java:349]: what's the difference between mParams.getQuery() & mParams.getGalSearchQuery(), is there an existing query param?

Comment (itsmeprashant) [reply store/src/java/com/zimbra/cs/gal/GalSearchControl.java:349]: So, SearchGal searches data on `name` request param, and mParams.getQuery(), holds the value for the same. Which searches inside the contact field.
And we want to search data inside any field, so we introduce a new parameter named `query`, and mParams.getGalSearchQuery() holds value for the same.

**Review by sreejan-chowdhury — APPROVED** [review]

File: store/src/java/com/zimbra/cs/service/account/SearchGal.java [hunk 2]
```diff
@@ -71,14 +71,27 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
...
         
-        // otherwise require a query
+        String galSearchQuery = null;
+        
         String name = null;
-        if (ref == null) {
-            name = request.getAttribute(AccountConstants.E_NAME);
-        }
+        
+		if (ref == null) {
```
Comment (dasiyogesh) [inline store/src/java/com/zimbra/cs/service/account/SearchGal.java:81]: indentation looks off for this if block, please correct it.

**Review by dasiyogesh — APPROVED** [review]

File: store/src/java/com/zimbra/cs/service/account/SearchGal.java (hunk 2 above)
Comment (RahulMehraSynacor) [reply store/src/java/com/zimbra/cs/service/account/SearchGal.java]: done

**Pull request merged**